server/protocols/webadmin/webadmin-mailrepository/src/main/java/org/apache/james/webadmin/service/MailRepositoryStoreService.java

```diff
@@ -86,13 +87,21 @@ public MailRepository createMailRepository(MailRepositoryPath repositoryPath, St
     }
 
     public Optional<MailDto> retrieveMail(MailRepositoryPath path, MailKey mailKey, Set<AdditionalField> additionalAttributes) throws MailRepositoryStore.MailRepositoryStoreException, MessagingException, InaccessibleFieldException {
-        return fecthMail(path, mailKey)
-            .map(Throwing.function((Mail mail) -> MailDto.fromMail(mail, additionalAttributes)).sneakyThrow());
+        Optional<Mail> mail = fecthMail(path, mailKey);
```

Comment:
Can we fix while we are at it the typo of that method? `fecthMail` -> `fetchMail`? :)